### DISCUSSION TITLE
Transform/Setter for a property on insert?

### DISCUSSION THREAD

**Discussion by notflip** [prisma/prisma#19800] [Q&A] [answered]
Is it possible to define a setter, or transformation in the schema?

Sequelize has the following method to achieve this, does something similar exist in Prisma?

```
        datum_afdruk: {
            type: Sequelize.DATE,
            set: function (value) {
                const parsedDate = DateTime.fromFormat(value, 'ddMMyy').setZone('Europe/Brussels')
                this.setDataValue('datum_afdruk', parsedDate.toISODate())
            }
        },
```

**Comment by ludralph** [comment, marked as answer]
Hi @notflip 👋 

Thank you for raising this question. While this is not directly possible in the Prisma Schema, you can leverage the [Prisma Client Extension](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) for adding a `transform` or `setter` property on `insert`.

Do take a look at this [example](https://www.prisma.io/blog/client-extensions-preview-8t3w27xkrxxn#example-instance-methods) on adding `instance methods` and adapt it for your use case.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by notflip** [reply]
I'm using middleware now to solve this, I suppose both methods would work. Thanks!

**Reply by ludralph** [reply]
Yes both methods would work.
